exporter/awsxrayexporter/internal/translator/http.go

```diff
@@ -193,7 +208,12 @@ func constructServerURL(urlParts map[string]string) string {
 	if !ok {
 		scheme = "http"
 	}
-	port := ""
+
+	port, ok := urlParts[conventions.AttributeNetHostPort]
+	if !ok {
+		port = ""
+	}
+
```

Comment:
Why did this logic need to get extracted out of the nested conditional below?

Reply:
It is a bug fix. The bug is detected by test case `TestServerSpanWithSchemeNamePortTargetAttributesStable`

Reply:
I would have greatly preferred if you split this out into a separate PR, since it's not related to the overall changes. Can you explain what the bug is and why the fix is correct? Please also update the overview to explain before/after of this bugfix.

Reply:
We are lucky that url.full is required attribute in Http client span, so we don't have to build http url for http client span by client.address and client.port.
Only server.address and server.port are needed for building http url for http server span.